src/store/actions/authActions.js

```diff
@@ -25,11 +25,9 @@ export const login = (email, password, history) => {
       const { userId } = result.data;
       //console.log(userId);
       dispatch(loginOkay(userId));
-      history.push("/hacks");
+      return "success"
```

Comment:
you're mixing up redux and components a bit. The component shouldn't know or care what gets returned from your actions, and the status should be managed through your reducer.

The normal pattern is to have something like

```
export const loadThing = () => {
  return async (dispatch, getState) => {
    try {
      dispatch(doingThing() // sets status in thing reducer = "LOADING"
      const result = await doThing()
      dispatch(thingSuccessful(result)) // sets status in thing reducer = "SUCCESSFUL"
    } catch (error) {
      console.error("Failed to load thing", error)
      dispatch(thingFailed()) // sets status in thing reducer = "FAILED"
    }
  }
}
dispatch(loginOkay(userId));
```

In your case for login, you might change the statuses to things like "LOGGING IN", "LOGGED_IN", and "LOGIN_FAILED"

Reply:
I was unaware that an action can have multiple dispatches, Cheers!

Reply:
have to be careful about the distinction between actions and thunks. thunks are dispatched, and can dispatch zero or more actions. actions are just plain js objects that have a `type` field and any other relevent fields (`payload` is common, but doesn't have to be that).